### ISSUE TITLE
Do not make assumptions how a project will be built

### ISSUE THREAD

**Issue by vaijab** [UKHomeOffice/docker-nodejs#11]
Currently, ONBUILD copies everything to `/app`, which is too much to assume. https://github.com/UKHomeOffice/docker-nodejs/blob/master/Dockerfile#L13

A common practice to build node apps images is to add `package.json` first, then run `npm install` and then add the rest of the code. Currently that's not possible. Less assumptions about how a project is going to be built would help.

**Comment by PurpleBooth** [comment]
What's the problem that this would solve?

**Comment by vaijab** [comment]
Say I have something in my `.` I don't want to just magically add.

**Comment by allandegnan** [comment]
We probably don't want to copy Dockerfile, readme.md, code-of-conduct.txt, etc.  More importantly, we don't want .git going up.

**Comment by PurpleBooth** [comment]
@allandegnan That should be done in your Docker ignore. If you don't agree can you add a new issue to discuss it?

**Comment by PurpleBooth** [comment]
@vaijab Understood, I'll make a PR for it

**Comment by vaijab** [comment]
Great. Thanks.

**Comment by allandegnan** [comment]
@PurpleBooth 

While it can be done in .dockerignore, it makes more sense for there to be a "src" folder or something that doesn't need to be remembered to be ignored.  Further, there are too many .things that something at sometime will be missed.

**Comment by PurpleBooth** [comment]
@allandegnan Create a new issue for that.

**Comment by allandegnan** [comment]
@PurpleBooth why?  This is the same issue.

**Comment by PurpleBooth** [comment]
@allandegnan Create a new issue for it. All further discussion on this issue will be deleted

This issue is regarding bringing this container inline with the official docker onbuild container.

**Comment by PurpleBooth** [comment]
Okay, turns out this issue wasn't about bringing this container inline with the official docker container.

It is the nature of onbuild containers that you make some assumption as to the structure and composition of the project. NodeJS projects do not actually need a package.json, so in order to remove all assumptions we would have to not do any onbuild steps at all.

This is that the 'docker-nodejs-base' image was originally for.

If there is no appetite for a onbuild container, I'd happily just change this to a container that has just node on it for the next version.

**Comment by jon-shanks** [comment]
yep this relates to https://github.com/docker-library/docs/tree/master/node#nodeonbuild
